test/routing/ArrivalsRouterSpec.scala

```diff
@@ -48,49 +48,56 @@ class ArrivalsRouterSpec extends AnyFreeSpec with Matchers with OptionValues wit
     new FakeV2ArrivalsController()
   )
 
-  "when creating a Arrival Notification" - {
-    "with accept header set to application/vnd.hmrc.2.0+json (version two)" - {
+  for (
+    endpoint <- Seq(
+      routes.ArrivalsRouter.createArrivalNotification(),
+      routes.ArrivalsRouter.getArrival("123"),
+      routes.ArrivalsRouter.getArrivalsForEori()
+    )
+  )
+    s"${endpoint.method} ${endpoint.url}" - {
+      "with accept header set to application/vnd.hmrc.2.0+json (version two)" - {
 
-      val arrivalsHeaders = FakeHeaders(
-        Seq(HeaderNames.ACCEPT -> "application/vnd.hmrc.2.0+json", HeaderNames.CONTENT_TYPE -> "application/xml")
-      )
+        val arrivalsHeaders = FakeHeaders(
+          Seq(HeaderNames.ACCEPT -> "application/vnd.hmrc.2.0+json", HeaderNames.CONTENT_TYPE -> "application/xml")
+        )
 
-      "must route to the v2 controller and return Accepted when successful" in {
+        "must route to the v2 controller and return Accepted when successful" in {
 
-        val request =
-          FakeRequest(method = "POST", uri = routes.ArrivalsRouter.createArrivalNotification().url, body = <test></test>, headers = arrivalsHeaders)
-        val result = call(sut.createArrivalNotification(), request)
+          val request =
+            FakeRequest(method = "POST", uri = endpoint.url, body = <test></test>, headers = arrivalsHeaders)
```

Comment:
should not use endpoint.method here ?

Reply:
It doesn't really matter here -- we just need to put _something_ into the URI of the request. We're not performing a full blown integration test so it doesn't route on this.

Reply:
if we are not using $endpoint.method then basically it is pointless to create it ?

Reply:
Oh, you're talking about the `method`. I thought you're talking about the URI.